src/pages/EditorTeam/EditorTeam.tsx

```diff
@@ -11,104 +8,44 @@ import DeleteIcon from '@material-ui/icons/Delete';
 import DoneIcon from '@material-ui/icons/DoneAllTwoTone';
 import EditIcon from '@material-ui/icons/EditOutlined';
 import RevertIcon from '@material-ui/icons/NotInterestedOutlined';
-
 import React, { useState, useEffect, FC } from 'react';
 
 import { useSelector } from 'react-redux';
 
-import { APIPerson } from 'models/registry.models';
+import { Rows } from 'models/global.enum';
 
+import { APIPerson } from 'models/registry.models';
 import Navigation from 'shared/Navigation/Navigation';
 import { RootState } from 'store/models/rootstate.model';
 
 import { deleteTeamMember, editTeamMember } from '../../helpers/editing-db.handler';
 
-import NewTeamMember from './components/NewTeamMember';
-import styles from './EditorTeam.module.css';
-
+import SelectTeam from './components/SelectTeam';
+import { CustomTableCell } from './functions/newCell';
+import { useStyles } from './stylesTable';
 
-const useStyles = makeStyles(theme => ({
-  root: {
-    width: '100%',
-    marginTop: theme.spacing(3),
-    overflowX: 'auto'
-  },
-  table: {
-    minWidth: 650
-  },
-  selectTableCell: {
-    width: 60
-  },
-  tableCell: {
-    width: 130,
-    height: 40
-  },
-  input: {
-    width: 130,
-    height: 40
-  }
-}));
-
-interface IPerson extends APIPerson{
+export interface IPerson extends APIPerson {
   lp?: number;
   isEditMode?: boolean;
 }
 
-interface IProps {
-  row: IPerson
-  name: string;
-  onChange: any;
-}
-
-
-const CustomTableCell = ({ row, name, onChange }: IProps) => {
-  const classes = useStyles();
-  const { isEditMode } = row;
-  return (
-    <TableCell align="left" className={classes.tableCell}>
-      {isEditMode ? (
-        <Input
-          value={row[name]}
-          name={name}
-          onChange={e => onChange(e, row)}
-          className={classes.input}
-        />
-      ) : (
-        row[name]
-      )}
-    </TableCell>
-  );
-};
-
 const EditorTeam: FC = () => {
   const registry = useSelector((state: RootState) => state.income.registry);
   const [rows, setRows] = useState<IPerson[] | undefined>();
   const [team, setTeam] = useState<string>('');
-  const [previous, setPrevious] = useState<any>({});
   const classes = useStyles();
-  const [openNewMember, setOpenNewMember] = useState<boolean>(false);
   const [actualValue, setActualValue] = useState<IPerson>(
-    {name: '', surname:'', id: ''}
+    { name: '', surname:'', id: '', dateOfAdd: null }
```

Comment:
And what about dateOfDelete?

Reply:
Working great, type dateOfDelete is Date | null | undefined, są i don't have to declare this key in state.